### ISSUE TITLE
Feature: Be able to select components

### ISSUE THREAD

**Issue by Skycoder42** [miurahr/aqtinstall#52]
Each Qt platform/version package consists of the primary package, as well as a few extra modules (like webengine or technical previews). It would be great if one could select explicitly which of those to install.

**Comment by miurahr** [comment]
If anyone helps to show how extra modules archive file determined from download.qt.io download repository.
ex. https://download.qt.io/online/qtsdkrepository/mac_x64/desktop/qt5_5140/Updates.xml

**Comment by Skycoder42** [comment]
How about doing this dynamically. For example, if `5.13.2` `linux` and `desktop` are selected, the base package would be: `qt.qt5.5132.gcc_64`.
Now I want to install `qt.qt5.5132.qtcharts` (and thus `qt.qt5.5132.qtcharts.gcc_64`) as well. Simply add a command line parameter like `--extra-mods qtcharts` and you can generate the package id automatically. The scheme for that would be `qt.qt5.<version>.<package>.<platform>`.

Once you have the package id, it should be no different to download this as for any other component

**Comment by miurahr** [comment]
I've just checked a logic how package selected. It seems that most of the modules, primary and extra which can be installed are selected, except for description only entry, and debug_info.
```
            self.update_xml = ElementTree.fromstring(r.text)
            for packageupdate in self.update_xml.iter("PackageUpdate"):
                name = packageupdate.find("Name").text
                if name.split(".")[-1] != self.arch:
                    continue
                if name.split(".")[-2] == "debug_info":
                    continue
                if packageupdate.find("DownloadableArchives").text is None:
                    continue
                if name == "qt.qt5.{}.{}".format(qt_ver_num, self.arch) or name == "qt.{}.{}".format(qt_ver_num, self.arch):
                    # basic packages
                    pass
                else:
                    # optional packages: FIXME: check option whether install or not
                    pass
```

It seems webengine is also installed (at least download and extracted) as well as primary Qt package.

**Comment by miurahr** [comment]
We can see extra modules are installed or not in CI log.

Here is a recent CI log on linux. I found qtwebengine was installed here.

https://dev.azure.com/miurahr/github/_build/results?buildId=1060&view=logs&jobId=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&taskId=b55c4b2f-2573-5bc7-7fb5-4995653abfc4&lineStart=114&lineEnd=115&colStart=1&colEnd=1

**Comment by Skycoder42** [comment]
Yes, that seems about right :+1:  Do you accept pull requests? If yes, I would add some code for this as a PR in the next few days?

**Comment by miurahr** [comment]
You are welcome for PR!

**Comment by miurahr** [comment]
Now the update is released as v0.6a1.
